hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -2239,6 +2271,16 @@ private static List<HoodieColumnRangeMetadata<Comparable>> getFileStatsRangeMeta
     return readColumnRangeMetadataFrom(partitionPath, fileName, datasetMetaClient, columnsToIndex, maxBufferSize);
   }
 
+  public static Option<Schema> getTableSchemaFromCommitMetadata(HoodieCommitMetadata commitMetadata, HoodieTableConfig tableConfig) {
```

Comment:
again, lets avoid public methods. can you make it package private

Reply:
do we have UTs for this

Reply:
I didn't write this I just refactored 2 places using identical code. I can undo the change if it is too confusing for you